src/Tools/Source/RunTests/AssemblyScheduler.cs

```diff
@@ -44,6 +44,16 @@ internal AssemblyInfo(string assemblyPath, bool useHmtl)
 
     internal sealed class AssemblyScheduler
     {
+        /// <summary>
+        /// This is a test class inserted into assemblies to guard against a .NET desktop bug.  The tests
+        /// inside of it counteract the underlying issue.  If this test is included in any assembly it 
+        /// must be added to every partition to ensure the work around is present
+        /// 
+        /// https://github.com/dotnet/corefx/issues/3793
+        /// https://github.com/dotnet/roslyn/issues/8936
+        /// </summary>
+        private const string EventListenerGuardFullName = "Microsoft.CodeAnalysis.UnitTests.EventListenerGuard";
```

Comment:
No.  Open to ideas on protecting it.  Can't really see a way of doing it though.